modules/extension/xsd/xsd-gml3/src/test/java/org/geotools/gml3/simple/GeometryEncoderTestSupport.java

```diff
@@ -62,6 +62,15 @@ protected Document encode(GeometryEncoder encoder, Geometry geometry) throws Exc
 
     protected Document encode(GeometryEncoder encoder, Geometry geometry, String gmlId)
             throws Exception {
+        return encode(encoder, geometry, true, gmlId);
```

Comment:
If it helps you can navigate from geometry --> geometry factory --> coordinate sequence factory to determine what is going on with the data, hasZ() and hasM() methods have been provided.

Reply:
Thanks.